pkg/detectors/salesforce/salesforce.go

```diff
@@ -0,0 +1,118 @@
+package salesforce
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+const (
+	currentVersion = "58.0" // current Salesforce version
+)
+
+type Scanner struct {
+	client *http.Client
+}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	defaultClient = common.SaneHttpClient()
+	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
+	accessTokenPat = regexp.MustCompile(`\b00[a-zA-Z0-9]{13}![a-zA-Z0-9_.]{96}\b`)
+	instancePat    = regexp.MustCompile(`\bhttps://[0-9a-zA-Z-\.]{1,100}\.my\.salesforce\.com\b`)
+)
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{"salesforce"}
+}
+
+// FromData will find and optionally verify Salesforce secrets in a given set of bytes.
+func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
+	dataStr := string(data)
+
+	instanceMatches := instancePat.FindAllStringSubmatch(dataStr, -1)
+	tokenMatches := accessTokenPat.FindAllStringSubmatch(dataStr, -1)
+
+	for _, instance := range instanceMatches {
+		if len(instance) != 1 {
```

Comment:
what is this check for? (i have the same question about `token` below.)

Reply:
this is to check that we are picking up regex matches correctly.

if we have capturing groups, then instead of 

`instanceMatches: [[https://trufflesecurity2-dev-ed.develop.my.salesforce.com]]`

it'll return 

`idMatches:  [[id e2e897c7-0887-4523-835d-e75eeedcdc52 e2e897c7-0887-4523-835d-e75eeedcdc52]]`

like we do on the docusign detector - so we expect two results per match 
`id e2e897c7-0887-4523-835d-e75eeedcdc52`
and 
`e2e897c7-0887-4523-835d-e75eeedcdc52` - so we use the second one with index [1]
 https://github.com/trufflesecurity/trufflehog/blob/c9f58b37808647936b3fb5fa230a9547ba96f3e1/pkg/detectors/docusign/docusign.go#L47
 
 tldr: it mostly checks if it's empty or otherwise not what we were expecting the regex to pick up.